src/sentry/rules/processing/processor.py

```diff
@@ -302,6 +334,39 @@ def apply_rule(self, rule: Rule, status: GroupRuleStatus) -> None:
                 )
                 return
 
+        predicate_func = get_match_function(condition_match)
+        if condition_list:
+            if not predicate_func:
+                log_string = f"Unsupported condition_match {condition_match} for rule {rule.id}"
+                logger.error(
+                    log_string,
+                    filter_match,
+                    rule.id,
+                    extra={**logging_details},
+                )
+                return
+            predicate_iter = (self.condition_matches(f, state, rule) for f in condition_list)
+            if predicate_iter:
+                result = predicate_func(predicate_iter)
+
+                if condition_match == "any" and not result:
+                    if slow_conditions and process_slow_conditions_later:
+                        self.enqueue_rule(rule)
+                        return
+                    return
+
+                elif condition_match == "all":
+                    if not result:
+                        return
+
+                    if slow_conditions and process_slow_conditions_later:
+                        self.enqueue_rule(rule)
+                        return
+
+        if slow_conditions and process_slow_conditions_later:
+            self.enqueue_rule(rule)
+            return
+
```

Comment:
We should remove this along with the other changes I mentioned

Reply:
We need this because of how we define `condition_list`:

```
        condition_list = fast_conditions
        if not process_slow_conditions_later:
            condition_list = fast_conditions + slow_conditions
```

if we have the flag, we need to enqueue the rule by this point